### PR TITLE
Add missing --log-driver and --log-opt docs

### PR DESCRIPTION
**Pull request by thaJeztah**
Adds (basic) documentation for "--log-driver" and "--log-opt" for services.

Also updated the API docs to include the new options, and generated a more complete JSON example.

relates to https://github.com/docker/docker/pull/24476

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @stevvooe @vdemeester ptal

**Comment by thaJeztah** [comment]
oh for future self, used this to create the request;

```
docker service create \
  --name web \
  --log-driver=json-file \
  --log-opt max-file=3 \
  --log-opt max-size=10M \
  --limit-memory=100M \
  --restart-condition=on-failure \
  --restart-delay=10s \
  --restart-max-attempts=10 \
  --replicas=4 \
  --update-delay=30s \
  --update-parallelism=2 \
  --label foo=bar \
  --mode=replicated \
  --mount type=volume,source=web-data,target=/usr/share/nginx/html,volume-label=com.example.something=something-value,readonly=true \
  --publish=8080:80 \
  nginx:alpine
```

**Comment by vdemeester** [comment]
LGTM 🐸

**Comment by thaJeztah** [comment]
author LGTM

File: docs/reference/api/docker_remote_api_v1.24.md [hunk 1]
```diff
@@ -3926,33 +3926,68 @@ Create a service
...
       "TaskTemplate": {
         "ContainerSpec": {
-          "Image": "redis"
+          "Image": "nginx:alpine",
+          "Mounts": [
+            {
+              "ReadOnly": true,
+              "Source": "web-data",
+              "Target": "/usr/share/nginx/html",
+              "Type": "VOLUME",
```
Comment (stevvooe) [inline docs/reference/api/docker_remote_api_v1.24.md]: This should be lowercase... 

https://github.com/docker/engine-api/blob/master/types/swarm/container.go#L25

**Comment by stevvooe** [comment]
@thaJeztah Thanks!

I swear I was going to do this today...

File: docs/reference/api/docker_remote_api_v1.24.md (hunk 1 above)
Comment (thaJeztah) [inline docs/reference/api/docker_remote_api_v1.24.md]: @stevvooe weird, as I copied this output from the daemon response, did that change recently?

**Comment by thaJeztah** [comment]
@stevvooe no worries, spreading the load :smile:

File: docs/reference/api/docker_remote_api_v1.24.md (hunk 1 above)
Comment (stevvooe) [inline docs/reference/api/docker_remote_api_v1.24.md]: Shouldn't have.

**Comment by thaJeztah** [comment]
@stevvooe looks like there's a bug then somewhere, just tried it again on master, and I see this appearing in the daemon (debug) log;

```
      "Mounts": [
        {
          "ReadOnly": true,
          "Source": "web-data",
          "Target": "/usr/share/nginx/html",
          "Type": "VOLUME",
          "VolumeOptions": {
            "DriverConfig": {
            },
            "Labels": {
              "com.example.something": "something-value"
            }
          }
        }
```

**Comment by stevvooe** [comment]
@thaJeztah File a bug. P1. This has to be fixed.

**Comment by thaJeztah** [comment]
@stevvooe I think I see the issue

**Comment by SvenDowideit** [comment]
this also looks to already be in bump_v1.12.0

**Pull request merged**